### PR TITLE
Add Sobel benchmark

### PR DESCRIPTION
**Pull request by KimNewell**
## Description
Add Sobel benchmark

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [x] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by jni** [comment]
@scikit-image/core for context, I helped @KimNewell set up this PR at an open source development workshop. Building up our benchmark suite seems like an easy win. ;) Based on @emmanuelle's benchmarks at https://github.com/scisprints/2018_05_sklearn_skimage_dask/issues/10#issuecomment-392356532, this benchmark should take about 0.5s to run.

File: benchmarks/benchmark_filters.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# See "Writing benchmarks" in the asv docs for more information.
```
Comment (stefanv) [inline benchmarks/benchmark_filters.py]: Which asv docs is this referring to?

**Comment by jni** [comment]
@stefanv thanks to @TomAugspurger, benchmarks are run nightly on master and results appear here:

http://pandas.pydata.org/speed/

See [here](https://github.com/scikit-image/scikit-image/pull/3137#issuecomment-396434513) for a discussion about running benchmarks before/after on each PR. Summary: not possible yet, but it should be possible to write a bot that does this. A reviewer-triggered bot is the right approach, because we don't want to overwhelm the benchmarks machine with every push. Rather, we should benchmark only when PRs are ready to be merged.

File: benchmarks/benchmark_filters.py (hunk 1 above)
Comment (jni) [reply benchmarks/benchmark_filters.py]: @stefanv good point! They are here:

http://asv.readthedocs.io/en/latest/writing_benchmarks.html

@KimNewell would you mind adding this line to your PR:

```
# http://asv.readthedocs.io/en/latest/writing_benchmarks.html
```

You edit the file to add that line and then `git commit -a` and `git push`, and this PR will be automatically updated. 😊

**Comment by jni** [comment]
Dang, that's a messed up push. Should have rebased instead of listening to GitHub's instructions.

**Comment by hmaarrfk** [comment]
@jni I think you can still rebase. In fact, I learned that as I was quickly deprecating PRs ;) 

After rebase:

```
commit 6bb0905bc2e11741b64e1cc774724c4e6cf7295c (HEAD -> sobel-benchmark)
Author: Juan Nunez-Iglesias <juan.n@unimelb.edu.au>
Date:   Wed Jun 27 15:56:01 2018 +1000

    Add support for peak memory tracking by SLIC

commit 88508252381a998347f32bb38f0d644c96bfedf6
Author: Kim Newell <Kim.Newell@csiro.au>
Date:   Wed Jun 20 11:01:44 2018 +1000

    Added link to asv docs

commit 777a106ac7c210fae2280ad319be959c4582facf
Author: Kim Newell <Kim.Newell@csiro.au>
Date:   Tue Jun 19 16:27:03 2018 +1000

    Add benchmark for filters.sobel

commit 2ce3c13c63c17e659d059e901931c9d55d1865a5
Author: Kim Newell <Kim.Newell@csiro.au>
Date:   Tue Jun 19 16:19:02 2018 +1000

    Renamed generic benchmark file to benchmark_segmentation

commit ddfec94351c3e1ccc6f5a79d41acaacea570b8af (upstream/master, master)
Author: Mark Harfouche <mark.harfouche@gmail.com>
Date:   Tue Jun 26 22:45:41 2018 -0700

    Fix the human readable error message on a bad build. (#3223)
```

**Comment by jni** [comment]
@hmaarrfk of course! Genius! 😅 

@stefanv @soupault @TomAugspurger can you have a look now?

**Comment by jni** [comment]
The Travis failures are related to `--pre` and not this PR.

**Comment by jni** [comment]
@stefanv sorry, I should have been more explicit in my docstring for `peakmem_setup`. Fixed. (The idea is that peakmem includes memory used by the setup function. By running a peak memory benchmark with no calls, we can measure how much memory setup takes, and therefore how much *more* memory the other function calls take.

http://asv.readthedocs.io/en/stable/writing_benchmarks.html#peak-memory

**Comment by jni** [comment]
Errrm, it's fixed in #3249. For some reason I can no longer push to @KimNewell's branch. Closing this PR in favour of #3249, which contains the same commit history.

**Pull request closed without merging**